lib/HLSL/DxilPromoteResourcePasses.cpp

```diff
@@ -184,11 +185,18 @@ bool DxilPromoteStaticResources::PromoteStaticGlobalResources(
       GlobalVariable *GV = *(it++);
       // Build list of instructions to promote.
       for (User *U : GV->users()) {
-        Instruction *I = cast<Instruction>(U);
-        Insts.emplace_back(I);
+        if (isa<LoadInst>(U) || isa<StoreInst>(U)) {
+          Insts.emplace_back(cast<Instruction>(U));
+        } else if (GEPOperator *GEP = dyn_cast<GEPOperator>(U)) {
+          for (User *gepU : GEP->users()) {
+            DXASSERT_NOMSG(isa<LoadInst>(gepU) || isa<StoreInst>(gepU));
```

Comment:
this will silently be ignored in a NoAsserts build. Will breaking this assumption cause any other failures? If so, compiler users may se a runtime crash without any actionable error messages...

Reply:
Do you think it's better if we don't add it, so promotion fails with:
`error: local resource not guaranteed to map to unique global resource.`
It's difficult to think of an actionable error message if this assumption fails.

At least this will crash in fewer cases now than it did before.